app/services/publisher_promo_token2018q1_generator.rb

```diff
@@ -0,0 +1,15 @@
+# Generates a token for a publisher for 2018q1 promotion
+class PublisherPromoToken2018q1Generator < BaseService
+  attr_reader :publisher
+  
+  def initialize(publisher:)
+    @publisher = publisher
+  end
+
+  def perform
+    return if Rails.application.secrets[:active_promo_id].blank?
```

Comment:
make sure we don't accidentally overwrite existing promo tokens

consider making this service omnipotent by adding a check for publisher.promo_token_2018q1
if it's present, return nil early

you could add a `force:` param to allow manual overwriting